### PR TITLE
#4 Fuzzy search, levenshtein distance float value deprecated

### PR DESCRIPTION
**Pull request by oligus**

### CODE REVIEW

File: src/Minimalcode/Search/Criteria.php [hunk 1]
```diff
@@ -404,18 +404,24 @@ public function notOperator()
      * Crates new predicate with trailing ~ optionally followed by levensteinDistance.
      *
      * @param string $value
-     * @param float $levenshteinDistance optional
+     * @param int $levenshteinDistance optional
```
Comment (dewos) [inline src/Minimalcode/Search/Criteria.php]: the notation should be float|int

File: src/Minimalcode/Search/Criteria.php [hunk 2]
```diff
@@ -404,18 +404,24 @@ public function notOperator()
...
+
+        /** Float deprecated in Solr */
+        if(is_float($levenshteinDistance)) {
+            $this->predicates[] = $this->processValue($value) . '~' . ($levenshteinDistance === null ? '' : $this->processFloat($levenshteinDistance));
+            return $this;
         }
 
         $this->predicates[] = $this->processValue($value) . '~' .
-            ($levenshteinDistance === null ? '' : $this->processFloat($levenshteinDistance));
+            (is_null($levenshteinDistance) ? '' : (int) $levenshteinDistance);
```
Comment (dewos) [inline src/Minimalcode/Search/Criteria.php]: is_null is slower (function call)

File: src/Minimalcode/Search/Criteria.php [hunk 3]
```diff
@@ -404,18 +404,24 @@ public function notOperator()
...
     {
-        if ($levenshteinDistance !== null && ($levenshteinDistance < 0 || $levenshteinDistance > 1)) {
-            throw new InvalidArgumentException('Levenshtein Distance has to be within its bounds (0.0 - 1.0).');
+        if ($levenshteinDistance !== null && ($levenshteinDistance < 0)) {
+            throw new InvalidArgumentException('Levenshtein Distance has to be 0 or above');
+        }
+
+        /** Float deprecated in Solr */
+        if(is_float($levenshteinDistance)) {
+            $this->predicates[] = $this->processValue($value) . '~' . ($levenshteinDistance === null ? '' : $this->processFloat($levenshteinDistance));
```
Comment (dewos) [inline src/Minimalcode/Search/Criteria.php]: Maybe you can simply process $levenshteinDistance with a ternary, no need of an extra block-if

$levenshteinDistance= is_float($levenshteinDistance) ? ...

also if() do not follow the prs2 (space required)

**Review by dewos — COMMENTED** [review]
@oligus hi, any chance to fix this pull request? Thanks

**Comment by oligus** [comment]
Thanx for the review comments!

**Comment by dewos** [comment]
Hi @oligus , thanks for your time, changes look good. Sadly with the new psr4 flat strucure (in master), the merge request has some conflicts. I'm sorry for that. Could you please rebase the changes in the new structure?  Thanks

**Comment by oligus** [comment]
Great, then I can try that out as well! I'm not sure I did it correctly though, but I merged your changes to my repo and then pushed.

**Comment by dewos** [comment]
+1 thanks @oligus

**Comment by oligus** [comment]
Thank you too @dewos !

**Pull request merged**